### PR TITLE
Add comment/uncomment.

### PR DESCRIPTION
**Pull request by KirillOsenkov**

### CODE REVIEW

**Comment by KirillOsenkov** [comment]
image
=>
image

**Comment by KirillOsenkov** [comment]
XmlComment

**Comment by KirillOsenkov** [comment]
Big thanks to @grabalon for helping me figure out the MultiSelectionBroker

File: Core/Dom/XmlDomExtensions.cs [hunk 1]
```diff
@@ -208,5 +208,34 @@ public static IEnumerable<T> SelfAndParentsOfType<T> (this XObject obj)
...
+
+			if (node.Span.Contains (span)) {
+				return node;
+			}
+
+			return null;
+		}
+
+
+		public static void VisitSelfAndChildren (this XNode node, Action<XNode> action)
```
Comment (mhutch) [inline Core/Dom/XmlDomExtensions.cs]: IMO this should be `VisitSelfAndDescendents`

File: Editor/Commands/CommentUncommentCommandHandler.cs [hunk 2]
```diff
@@ -0,0 +1,500 @@
...
+					end = i + 1;
+					break;
+				}
+			}
+
+			return TextSpan.FromBounds (start.Value, end);
+
+			bool IsWhiteSpace (char c)
+			{
+				return c == ' ' || c == '\t' || (c > (char)128 && char.IsWhiteSpace (c));
```
Comment (mhutch) [inline Editor/Commands/CommentUncommentCommandHandler.cs]: https://github.com/mhutch/MonoDevelop.Xml/blob/8bf888f8e722384b9776e424311bf7e83cff2d7a/Core/Parser/XmlChar.cs#L40

**Comment by mhutch** [comment]
This looks great!

File: Core/Dom/XmlDomExtensions.cs (hunk 1 above)
Comment (KirillOsenkov) [reply Core/Dom/XmlDomExtensions.cs]: TIL descendants vs. descendents. Seeing that you're using descendents everywhere I will use that too

**Comment by KirillOsenkov** [comment]
Fixed!

**Pull request merged**